src/components/Icon/FontIcon/FontIcon.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { AriaRole, forwardRef } from "react";
+import classNames from "classnames";
+import VibeComponentProps from "../../../types/VibeComponentProps";
+import VibeComponent from "../../../types/VibeComponent";
+
+interface FontIconProps extends VibeComponentProps {
+  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+  "aria-label"?: string;
+  tabIndex?: number;
+  icon?: string | React.Component | null;
+  role?: AriaRole;
+  "aria-hidden"?: boolean;
+}
+
+const FontIcon: VibeComponent<FontIconProps, HTMLElement> = forwardRef(
+  (
+    {
+      id = "",
+      className,
+      onClick,
+      "aria-label": iconLabel,
+      tabIndex,
+      icon: Icon,
+      role = "img",
+      "aria-hidden": ariaHidden,
+      "data-testid": dataTestId
+    },
+    iconRef
+  ) => {
+    const isIconFunction = typeof Icon === "function";
+    const iconClassName = isIconFunction ? "" : Icon;
+    // @ts-ignore
+    return (
+      // eslint-disable-next-line jsx-a11y/click-events-have-key-events
+      <span
+        aria-hidden={ariaHidden}
+        className={classNames(className, "fa", iconClassName)}
+        onClick={onClick}
+        ref={iconRef}
+        aria-label={iconLabel}
+        tabIndex={tabIndex}
+        role={role}
+        id={id}
+        data-testid={dataTestId}
+      >
+        {isIconFunction && (
+          //Icon components are not converted to typescript in this stage
+          //@ts-ignore
```

Comment:
Why?

Reply:
as I wrote in the comment:           //Icon components are not converted to typescript in this stage
this is why